predicators/datasets/vlm_input_data.py

```diff
@@ -340,3 +568,140 @@ def create_ground_atom_data_from_labeled_txt(
     return Dataset(low_level_trajs, ground_atoms_trajs)
+
+
+def create_ground_atom_data_from_img_trajs(
+        env: BaseEnv,
+        train_tasks: List[Task],
+        known_options: Set[ParameterizedOption],
+        vlm: Optional[VisionLanguageModel] = None) -> Dataset:
+    """Given a folder containing trajectories that have images of scenes for
+    each state, as well as options that transition between these states, output
+    a dataset."""
+    trajectories_folder_path = os.path.join(CFG.data_dir,
+                                            CFG.vlm_trajs_folder_name)
+    # First, run some checks on the folder name to make sure
+    # we're not accidentally loading the wrong one.
+    folder_name_components = CFG.vlm_trajs_folder_name.split('__')
+    assert folder_name_components[0] == CFG.env
+    assert folder_name_components[1] == "vlm_demos"
+    assert int(folder_name_components[2]) == CFG.seed
+    assert int(folder_name_components[3]) == CFG.num_train_tasks
+    num_trajs = len(os.listdir(trajectories_folder_path))
+    assert num_trajs == CFG.num_train_tasks
+    option_name_to_option = {opt.name: opt for opt in known_options}
+    image_option_trajs = []
+    all_task_objs = set()
+    for train_task_idx, path in enumerate(
+            sorted(Path(trajectories_folder_path).iterdir())):
+        assert path.is_dir()
+        state_folders = [f.path for f in os.scandir(path) if f.is_dir()]
+        num_states_in_traj = len(state_folders)
+        state_traj = []
+        for state_num in range(num_states_in_traj):
+            currimgs = []
+            curr_state_path = path.joinpath(str(state_num))
+            # NOTE: we assume all images are saved as jpg files.
+            img_files = glob.glob(str(curr_state_path) + "/*.jpg")
+            for img in img_files:
+                currimgs.append(PIL.Image.open(img))
+            state_traj.append(currimgs)
+        # Get objects from train tasks to be used for future parsing.
+        curr_train_task = train_tasks[train_task_idx]
+        curr_task_objs = set(curr_train_task.init)
+        all_task_objs |= curr_task_objs
+        curr_task_obj_name_to_obj = {obj.name: obj for obj in curr_task_objs}
+        # Parse out actions for the trajectory.
+        options_traj_file_list = glob.glob(str(path) + "/*.txt")
+        assert len(options_traj_file_list) == 1
+        options_traj_file = options_traj_file_list[0]
+        with open(options_traj_file, "r", encoding="utf-8") as f:
+            options_file_str = f.read()
+        option_names_list = re.findall(r'(\w+)\(', options_file_str)
+        parsed_str_objects = re.findall(r'\((.*?)\)', options_file_str)
+        object_args_list = [obj.split(', ') for obj in parsed_str_objects]
+        # Remove empty square brackets from the object_args_list.
+        for object_arg_sublist in object_args_list:
+            object_arg_sublist.remove('[]')
+        parameters = [
```

Comment:
have you tested this? if not, might as well just assert that there are no continuous parameters until we have a case where we need it (and are able to test it)

Reply:
i have tested it actually by just writing continuous params into the input txt file, so i prefer leaving it in for now